### DISCUSSION TITLE
Calling middleware next() results in type error

### DISCUSSION THREAD

**Discussion by raisonjoseph** [socketio/socket.io#4010] [General]
When using node `redis` and `socket.io `in the same project the middlewares for socket.io whenever calling `next()` without argument results in redis type error. I have tried out different version of redis and the issue starts after redis `v2.8.0`

There is no issue when throwing and error inside the next fucntion

```
/home/entri/Workspace/illustrations/crack/OneP-backend/node_modules/redis/index.js:859
            command_obj.callback(undefinedArgError);
                        ^

TypeError: command_obj.callback is not a function
    at RedisClient.internal_send_command (/node_modules/redis/index.js:859:25)
    at RedisClient.set (/node_modules/redis/lib/commands.js:46:25)
    at Namespace.emit (events.js:314:20)
    at Namespace.emitReserved (node_modules/socket.io/dist/typed-events.js:56:22)
    at node_modules/socket.io/dist/namespace.js:140:26
```

the code 

```
   io.use((socket, next) => {
        next()
    }).on('connect', (client) => {

    });
  
```
To recreate simply install socket.io and redis under same node server.

**Comment by darrachequesne** [comment]
Hi! I was not able to reproduce.

The stack trace suggests you have a Redis operation (`RedisClient.set`) in a `connection` handler (`Namespace.emit`) without a callback. Please note that the Redis client provided by the `redis` module does not support Promises, you have to pass a callback to each method:

```js
redisClient.set("mykey", "myvalue", () => { /* ... */ });
```

Reference: https://github.com/NodeRedis/node-redis

Could you please check?
